catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/validation/ValidateExecutor.java

```diff
@@ -0,0 +1,72 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog.validation;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.stream.Collectors;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+public class ValidateExecutor {
+
+    public static List<ValidateReport> execute(List<Metacard> metacards,
+            List<MetacardValidator> validators) {
+
+        ForkJoinPool forkJoinPool = new ForkJoinPool();
+        List<ValidateReport> results;
+        try {
+            results = forkJoinPool.submit(() -> metacards.stream()
+                    .parallel()
+                    .map(metacard -> generateReport(metacard, validators))
+                    .collect(Collectors.toList()))
+                    .get();
+        } catch (InterruptedException | ExecutionException ee) {
```

Comment:
I was considering just having the method throwing the exception instead, and in hindsight I might like that better so that the code calling this function can be better aware of why a validation failed. Thoughts?